lib/token.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const BaseModel = require('./base');
+const nodeify = require('./nodeify');
+const iron = require('iron');
+
+// Symbols for private members
+const password = Symbol('password');
+
+class TokenModel extends BaseModel {
+    /**
+     * Construct a TokenModel object
+     * @method constructor
+     * @param  {Object}    config
+     * @param  {Object}    config.datastore         Object that will perform operations on the datastore
+     * @param  {Number}    config.userId            The ID of the associated user
+     * @param  {String}    config.value             The token value
+     * @param  {String}    config.tokenId           The token ID
+     * @param  {String}    config.name              The token name
+     * @param  {String}    config.description       The token description
+     * @param  {Date}      config.lastUsed          The last time the token was accessed
+     * @param  {String}    config.password          The encryption password
+     */
+    constructor(config) {
+        super('token', config);
+        this[password] = config.password;
+    }
+
+    /**
+     * Update a token with sealed token value
+     * @method update
+     * @return {Promise}
+     */
+    update() {
+        return nodeify.withContext(iron, 'seal', [this.value, this[password], iron.defaults])
```

Comment:
> it isn't really the responsibility of the model to determine access control

If the concern is exposing too much to the caller, an alternative would be to provide another method that compares a given value to the token. Something like:

```
const myPassedInValue = 'maybe_the_right_secret';

const match = token.compare(myPassedInValue);

console.log(match); // true if the value matches the token's value, false otherwise
```

The methods defined by the base model are the essential methods for synchronizing a model's data to the datastore. If there are other business rules that a particular model is responsible for, you can create additional methods to handle that case.